numpy/lib/function_base.py

```diff
@@ -4126,6 +4110,35 @@ def percentile(a, q, axis=None, out=None,
     else:
         return r
 
+def _median_nancheck(data, result, axis, out):
+    """
+    Check median result from data for NaN values at the end and return NaN
+    in that case. Also used for masked median
+    """
+    if data.size == 0:
+        return result
+    data = np.rollaxis(data, axis, data.ndim)
+    n = np.isnan(data[..., -1])
+    # masked NaN are fine
+    try:
+        n = n.filled(False)
```

Comment:
This is to deal with the possibility that n is not masked, no? Either a comment or explicit type check would help. Should also be mentioned in the docstring that it handles both.